### ISSUE TITLE
API is unclear

### ISSUE THREAD

**Issue by nickdaugherty** [dominictarr/reconnect#8]
From the documentation, it's difficult to determine exactly how to use this plugin.

The docs mention passing the `stream` to `reconnect()`, but also passing `opts` and `onConnect` - how is it actually used?

A more intuitive and standard API would be:

```
reconnect(stream, options);
```

**Comment by dominictarr** [comment]
Thanks for posting a documentation issue!

reconnent needs know how to create a new connection if the current connection ends,

so the api is

```
reconnect(function (stream) {
  //your code here
}, options)
```

`reconnect(stream, options)`, wouldn't work, because reconnect wouldn't know what sort of stream you have.

what sort of connection are you trying to reconnect?

**Comment by nickdaugherty** [comment]
Makes sense...so is `reconnect (opts, onConnect)` (from the docs) a valid function signature, or is that backwards?

**Comment by dominictarr** [comment]
oh, right - I gave the options in the other order in my example there - that should have been the other way around,
or you can leave options out and get the defaults (which is what I normally do)
